apps/src/code-studio/progressRedux.js

```diff
@@ -208,116 +227,54 @@ export default function reducer(state = initialState, action) {
         viewType === ViewType.Student && state.studentDefaultsSummaryView
     };
   }
-
   if (action.type === SET_CURRENT_STAGE_ID) {
     // if we already have a currentStageId, that means we're on a puzzle page,
     // and we want currentStageId to remain the same (rather than reflecting
     // the last stage the user has made progress on).
     if (state.currentStageId) {
       return state;
     }
-
     return {
       ...state,
-      currentStageId: action.stageId
+      currentStageId: action.stageId,
+      currentLesson: getLesson(state, action.stageId)
     };
   }
-
   if (action.type === SET_SCRIPT_COMPLETED) {
     return {
       ...state,
       scriptCompleted: true
     };
   }
-
   if (action.type === SET_STAGE_EXTRAS_ENABLED) {
     return {
       ...state,
       stageExtrasEnabled: action.stageExtrasEnabled
     };
   }
-
   return state;
 }
 
 // Helpers
 
-/**
- * Return the level with the highest progress, or the first level if none have
- * been attempted
- * @param {number[]} levelIds
- * @param {Object.<number,number>} - Mapping from level id to progress result
- */
-function bestResultLevelId(levelIds, progressData) {
```

Comment:
I noticed while working through the tests that we no longer use this concept of "bestResultLevelId" - why were we able to get rid of it?

Reply:
TBH i don't fully understand the consequences of this, but when we discussed it in the live code review it sounded like there was general agreement that this is safe to remove. i've added it to the list of things we should explicitly check in the next bug bash though.

here's my best understanding of what this was for any why it's safe to remove. @dmcavoy lmk if you have anything to add or correct!

level progress is always associated with a particular `Level` model object. however, the `levels` we get in our script summaries from the backend are actually from `ScriptLevel`s, not `Level`s. a `ScriptLevel` is "sort-of-but-not-quite a join table between Scripts and Levels" (from the comment in the model file), and has_and_belongs_to_many :levels. my understanding is that the levels in that association might contain inactive versions of the level or (deprecated) level variants. so i believe the purpose of this function was to make sure that we show the best progress the user has achieved on this script level, even if it's not progress on the currently active level for that script level.

i think we no longer need this for two reasons:
1) i think it mostly relates to legacy functionality, but i'm still working on confirming that
2) i'm not convinced it makes sense to show progress on a level that's not the currently active level in the assigned script, but again, i'm not positive about this due to lack of context.

let's be sure to chat about this in Tuesday's bug bash!